### ISSUE TITLE
[Tutorial A11y]: Issues reading Make It So Tutorial using VoiceOver, clicking on tutorial steps to show related code blocks is inaccessible

### ISSUE THREAD

**Issue by robin24** [peterfriese/MakeItSo#121]
First off, many thanks to you for having created this absolutely great tutorial!

I am a fully blind developer, and wanted to get started on learning how to integrate Firebase with SwiftUI. Sadly there don't seem to be too many detailed text tutorials out there, most content only exists in video format which is hard for me to follow.

As a MacOS user, I'm using [VoiceOver](https://www.apple.com/accessibility/features/?vision), the screenreader built into all of Apple's platforms.
While the Make It So tutorial page can be navigated and read with VoiceOver without any issues, there unfortunately is a major accessibility bug when it comes to showing the code snippets for the various tutorial steps.
From what I understand, normally a user would click on (or maybe just hover over?) each of the steps that have associated code blocks, which would then show the code block associated with that step.
Unfortunately, it seems like some markup might be missing that would allow a screenreader like VoiceOver to detect these as clickable elements, basically making it impossible to show or hide the associated code blocks.

If you could kindly look into this when you have some time, I'd super greatly appreciate that. Like I said, based on my own findings this appears to be the most in-depth text tutorial on getting up to speed with Firebase and SwiftUI, and I'd love to go through the experience of building the Make It So app from scratch.

In case fixing the tutorial page requires too much work or isn't feasible for any reason, I'd also be happy about a text version of the tutorial.

Thanks, and keep up the great work!

**Comment by peterfriese** [comment]
Hey @robin24, thanks so much for the kind words. I really appreciate it!

I'm sorry to hear that there are accessibility issues with the tutorial. Have you tried Apple's tutorials? Do they have the same issue? If not, then there is definitely a way to fix this. It'll just take some time...

**Comment by robin24** [comment]
Hey @peterfriese,

Thanks for your response, you're more than welcome!

Now that you've mentioned Apple's tutorials, I remembered [This one for the Swift OpenAPI Generator](https://swiftpackageindex.com/apple/swift-openapi-generator/1.7.2/tutorials/swift-openapi-generator/clientxcode).
This seems to have a very similar layout to your tutorial, and actually happens to have the exact same accessibility issue...
So now I'm curious, what's the name of the underlying tool that is used for page generation here? Perhaps it would be a good idea to open an issue about this in that project's repo for an upstream fix.

**Comment by peterfriese** [comment]
Hi @robin24 - I've reached out to the DocC team, and they confirmed this is an issue in the renderer. 

This is now being tracked at https://github.com/swiftlang/swift-docc-render/issues/949

Once this has been resolved, it should be sufficient to rebuild the tutorial to get proper A11y support.

Thanks again for raising this!

**Comment by robin24** [comment]
Hey @peterfriese,

Thanks so much for taking the time to reach out to them with the issue, super greatly appreciated!
Hopefully this should improve lots of other doc pages / tutorials as well once there is a fix.
Meanwhile, I've found that saving individual tutorial pages to disk and then reading them with TextMate is quite doable as a workaround :-).
